shared/settings/chat/index.tsx

```diff
@@ -70,9 +93,102 @@ class Chat extends React.Component<Props, State> {
     this.props.onRefresh()
   }
 
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.contactSettingsEnabled !== prevProps.contactSettingsEnabled) {
+      this.setState({contactSettingsEnabled: this.props.contactSettingsEnabled})
+    }
+    if (this.props.contactSettingsIndirectFollowees !== prevProps.contactSettingsIndirectFollowees) {
+      this.setState({contactSettingsIndirectFollowees: this.props.contactSettingsIndirectFollowees})
+    }
+    // Create an initial copy of teams data into state, so it can be mutated there.
+    if (
+      Object.keys(this.props.contactSettingsSelectedTeams).length > 0 &&
+      Object.keys(this.state.contactSettingsSelectedTeams).length == 0
+    ) {
+      this.setState({contactSettingsSelectedTeams: this.props.contactSettingsSelectedTeams})
+    }
+  }
```

Comment:
This comment is more of an exploration of whether I understand hooks correctly, and less a serious suggested change.

Could this be a functional component with hooks? `useState` and `useEffect` should work here?


```js
render() {
    const [contactSettingsEnabled, setContactSettingsEnabled] = React.useState(undefined)
    const [contactSettingsIndirectFollowees, setSontactSettingsIndirectFollowees] = React.useState(undefined)
    const [contactSettingsTeamsEnabled, setSontactSettingsTeamsEnabled] = React.useState(undefined)
    const [contactSettingsSelectedTeams, setContactSettingsSelectedTeams] = React.useState({})
    React.useEffect(() => {
        setContactSettingsSelectedTeams(props.contactSettingsSelectedTeams)
    }, [Object.keys(props.contactSettingsSelectedTeams).length > 0
        && Object.keys(contactSettingsSelectedTeams).length == 0])
    return (
        ...
        // Later - can use the `prevState` function
        <TeamRow
            ...
            onCheck={(checked: boolean) => setContactSettingsSelectedTeams(
                prevState => ({ ...prevState, [teamDetails.id]: checked })
            }
        />
    )
}
```

Reply:
Looks right!  I thought I'd keep it in line with the way Unfurl works too.